salt/modules/salt_check.py

```diff
@@ -0,0 +1,449 @@
+#!/usr/bin/env python
+# -*- coding: UTF-8 -*-
+'''This custom salt module makes it easy to test salt states and highstates.
+   Author: William Cannon  william period cannon at gmail dot com'''
+import os
+import os.path
+import yaml
+import salt.client
+import salt.minion
+import salt.config
+
+
+class SaltCheck(object):
+    '''
+    This class implements the salt_check
+    '''
+
+    def __init__(self):
+        self.__opts__ = salt.config.minion_config('/etc/salt/minion')
```

Comment:
Thanks Mike.  I appreciate the info.  I'm open to solutions.....perhaps I
should detect the OS, and pull accordingly?



On Wed, Jun 1, 2016 at 12:02 PM, Mike Place <notifications@github.com>
wrote:

> In salt/modules/salt_check.py
> <https://github.com/saltstack/salt/pull/33642#discussion_r65401400>:
>
> > +   Author: William Cannon  william period cannon at gmail dot com'''
> > +import os
> > +import os.path
> > +import yaml
> > +import salt.client
> > +import salt.minion
> > +import salt.config
> > +
> > +
> > +class SaltCheck(object):
> > +    '''
> > +    This class implements the salt_check
> > +    '''
> > +
> > +    def __init__(self):
> > +        self.__opts__ = salt.config.minion_config('/etc/salt/minion')
>
> This won't work across different platforms, i.e. Windows. We need to find
> another way to get opts in here.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/saltstack/salt/pull/33642/files/3c72408d16a62590dc78015f835e5b7d06b86d59#r65401400>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA3k_VKaESgdvGH5p17382rJVqOhaKUwks5qHbspgaJpZM4IqzaF>
> .
>